### PR TITLE
fix testing, integration

### PR DESCRIPTION
**Pull request by pranathivemuri**
1. added github actions
2. makefile running tests coverage
3. linting using black, install using pip install black and do `black .` to lint and reformat the files automatically using black when you commit next time
4. pypi package upload (we can talk about how it would trigger on a call)
5.  correct tests to compare and check with the npz file keys
6. improved how all the for loops are run with itertools 
7. separated out the .npz files for regression coefficients

### CODE REVIEW

File: ortho_seq_code/orthogonal_polynomial.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def orthogonal_polynomial(
                 phi[j][i][2] = 1.0
 
     naming = os.path.basename(f.name)
-    if precomputed:
+    if precomputed == 'True':
```
Comment (pranathivemuri) [inline ortho_seq_code/orthogonal_polynomial.py:95]: this is not how you compare a boolean if precomputed already does that you can't compare a boolean to a string called True!

Comment (pranathivemuri) [reply ortho_seq_code/orthogonal_polynomial.py]: I am changing it back, please pull @snafees

File: conda_environment.yml [hunk 2]
```diff
@@ -5,4 +5,7 @@ channels:
   - defaults
 dependencies:
   - numpy
-  - Click
\ No newline at end of file
+  - Click
+  - pytest
+  - coverage
+  - flake8
```
Comment (phoenixAja) [inline conda_environment.yml:11]: picky, but can you alphabetize this? I think it looks nicer

File: ortho_seq_code/constants_orthoseqs.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "T",
+    "W",
+    "Y",
+    "V",
+]
+PROTEIN_ALPHABETS_N = PROTEIN_ALPHABETS + ["n"]
+PROTEIN_ALPHABETS_POLAR = ["R", "N", "D", "C", "E", "Q", "H", "K", "S", "T", "Y"]
+PROTEIN_ALPHABETS_NONPOLAR = ["A", "G", "I", "L", "M", "F", "P", "W", "V"]
+
+DM_ALPHABETS = {
```
Comment (phoenixAja) [inline ortho_seq_code/constants_orthoseqs.py:29]: i think this would be better as an enum class instead of a dict.

**Review by phoenixAja — APPROVED** [review]
Really nice work here! Definitely will be nice to have CI. I also really approve of adding a constants file. One minor thing is that next time it would be easier to review if you were only doing one thing at a time for each PR (ex: one PR to add CI, another to lint everything with black, and then another to add constants file, etc. ) that would make it a little easier to review. Awesome improvements!

**Review by snafees — APPROVED** [review]
Thanks! Else condition works and things are now getting saved correctly after I added some naming changes to ensure that file names better reflect what is being saved.

**Pull request merged**